administrator/components/com_associations/views/associations/tmpl/default.php

```diff
@@ -85,6 +85,7 @@
 			<tbody>
 			<?php foreach ($this->items as $i => $item) :
 				$canEdit    = $user->authorise('core.edit', $this->state->get('component') . $item->id);
+				
 				if (isset($item->created_by))
 				{
 					$canEditOwn = $user->authorise('core.edit.own', $this->state->get('component') . $item->id) && $item->created_by == $userId;
```

Comment:
replace all $this->state->get('component') by $this->component->component in this ACL rules

why? just make a dump of the vars and you will see.